### PR TITLE
Adjust FSM handling of non-broadcast

### PR DESCRIPTION
**Pull request by JohnOH**
Depending on how the calling sketch uses rf12_recvDone and thence rf69::recvDone_compat. It is possible to stall in TXIDLE after a uninteresting non-broadcast packet is *delay discarded* rather than discarded immediatley.

### CODE REVIEW

File: RF69.cpp [hunk 1]
```diff
@@ -161,22 +173,44 @@ uint8_t* recvBuf;
...
+        // It would be nice to wrap this code up and share it with identical
+        // code in case TXRECV below.
+        rxfill = rf12_len = 0;
         crc = _crc16_update(~0, group);
         recvBuf = buf;
         rxstate = TXRECV;
         flushFifo();
+        writeReg(REG_DIOMAPPING1, DMAP1_SYNCADDRESS);    // Interrupt trigger
         setMode(MODE_RECEIVER);
+        writeReg(REG_AFCFEI, AfcClear);
```
Comment (JohnOH) [inline RF69.cpp]: Dependencies on other registers RFxConsole. Drop

File: RF69.cpp [hunk 2]
```diff
@@ -161,22 +173,44 @@ uint8_t* recvBuf;
...
+        // It would be nice to wrap this code up and share it with identical
+        // code in case TXIDLE above.
+                    rxfill = rf12_len = 0;
+                    crc = _crc16_update(~0, group);
+                    recvBuf = buf;
+                    rxstate = TXRECV;
+                    flushFifo();                    
+                    writeReg(REG_DIOMAPPING1, DMAP1_SYNCADDRESS);// Interrupt trigger
+                    setMode(MODE_RECEIVER);
+                    writeReg(REG_AFCFEI, AfcClear);
```
Comment (JohnOH) [inline RF69.cpp]: Dependencies on other registers RFxConsole. Drop

**Comment by jcw** [comment]
I've merged a somewhat altered version of this code - I think the whole problem can be ascribed to "rxstate = TXIDLE" being set too late in the "case TXRECV:" branch of the switch.

**Comment by JohnOH** [comment]
Does this not have the same problem outcome for rf12_canSend() since the packet isn't fully discarded until a call into rf12_recvdone happens. I think this code still has the stall..

**Comment by jcw** [comment]
Ah, I see your point. But hasn't this always been the case: when recvDone returns true, you get a brief hold on the data, and things resume on the next recvDone call. The RF12 code is written under the assumption of lots of polling to keep the FSM moving.

Hm.. need to think more about this, am re-opening this for now.

**Comment by JohnOH** [comment]
Yes, I think it has always been the case but I don't think it needs to be. Once it comes out of TXRECV it has to deliver something because rxfill != 0 if nothing else.

**Pull request closed without merging**